modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -1481,13 +1495,30 @@ public void discardMediaPackage(MediaPackage mp) throws IOException {
     logger.info("Sucessful discarded mediapackage {}", mp);
   }
 
+  /**
+   * Creates a StandAloneTrustedHttpClientImpl
+   *
+   * @param user
+   * @param password
+   * @return
+   */
+  protected TrustedHttpClient createStandaloneHttpClient(String user, String password) {
+    return new StandAloneTrustedHttpClientImpl(this.downloadUser, this.downloadPassword, none(), none(), none());
+  }
+
   protected URI addContentToRepo(MediaPackage mp, String elementId, URI uri) throws IOException {
     InputStream in = null;
     HttpResponse response = null;
+    TrustedHttpClient httpClientStandAlone = httpClient;
     try {
       if (uri.toString().startsWith("http")) {
         HttpGet get = new HttpGet(uri);
-        response = httpClient.execute(get);
+
+        if (uri.getHost().matches(this.downloadSource)) {
```

Comment:
We `trimToNull` downloadSource, and the other configurables above.  Does it make sense to throw some kind of warning or error here, rather than silently failing?

Reply:
Yep right. But im am not sure where to put this in our opencast code world. From my perspective this null should be handled during variable allocation at the beginning? Or check all credentials user,pw,source right here and throw an error. What do you think?

Reply:
Personally I would do the sanity check when they're first parsed in `updated()`, and throw an (IIRC) `ConfigurationException` if they're invalid.  This fails the startup of the service, which prevents having to handle the case later :)